File: restartonempty/src/main/java/uk/asheiou/restartonempty/EventListener.java
package uk.asheiou.restartonempty;

import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;
import org.bukkit.plugin.java.JavaPlugin;

public class EventListener implements Listener {
  
  private final JavaPlugin plugin;
  
  public EventListener(JavaPlugin plugin) { this.plugin = plugin; }
  
  @EventHandler(priority = EventPriority.LOWEST)
  public void onPlayerQuit(PlayerQuitEvent event) {
    new ROEQuitTask().runTaskLater(this.plugin, 3);
  }
}

File: src/main/kotlin/cymru/asheiou/restartonempty/EventListener.kt
package cymru.asheiou.restartonempty

import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent
import org.bukkit.plugin.java.JavaPlugin

class EventListener(private val plugin: JavaPlugin) : Listener {

  @EventHandler(priority = EventPriority.LOWEST)
  fun onPlayerQuit(event: PlayerQuitEvent) {
    ROEQuitTask().runTaskLater(this.plugin, 3)
  }
}
